extensions/elasticsearch-rest-client-common/deployment/src/main/java/io/quarkus/elasticsearch/restclient/common/deployment/DevServicesElasticsearchProcessor.java

```diff
@@ -197,7 +197,7 @@ private DevServicesResultBuildItem.RunningDevService startElasticsearch(
         // Starting the server
         final Supplier<DevServicesResultBuildItem.RunningDevService> defaultElasticsearchSupplier = () -> {
             ElasticsearchContainer container = new ElasticsearchContainer(
-                    DockerImageName.parse(config.imageName));
+                    DockerImageName.parse(config.imageName).asCompatibleSubstituteFor("elasticsearch/elasticsearch"));
```

Comment:
We probably also need `elasticsearch` here (or is it `_/elasticsearch`? I mean the one with no prefix, see https://hub.docker.com/_/elasticsearch/ vs. https://hub.docker.com/elastic/elasticsearch/).

Ideally we'd also allow OpenSearch, since our stack *is* compatible with it (except for the high-level client, maybe). But that may be a bit more adventurous. Relevant images:

* `opensearchproject/opensearch` (see https://hub.docker.com/r/opensearchproject/opensearch)
* `public.ecr.aws/opensearchproject/opensearch` (see https://opensearch.org/docs/latest/opensearch/install/docker/)

Reply:
I'm not so sure. The whole purpose of it is to have `ElasticsearchContainer` to not complain about the custom image. And it is actually just checking that it's compatible with the default one.

We just want to pass:
`dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME, DEFAULT_OSS_IMAGE_NAME);`

Reply:
Ok, well if you think that's enough I'll trust you.

Reply:
That worked for me:
`DockerImageName.parse(config.imageName).asCompatibleSubstituteFor("docker.elastic.co/elasticsearch/elasticsearch"));`
Just My 2 Cents.